src/debugpy/_vendored/pydevd/conftest.py

```diff
@@ -106,13 +106,41 @@ def check_no_threads():
     _start_monitoring_threads()
 
 
+from _pydevd_bundle.pydevd_utils import is_current_thread_main_thread
+import threading
+
+
+@pytest.yield_fixture(autouse=True)
+def check_main_thread(request):
+    was_main = is_current_thread_main_thread()
```

Comment:
I'm still debugging this (but I wanted to merge what I already have anyways as this is proving to be quite hard to track down).

This fails only under Windows (and only sometimes) -- it seems that threading.current_thread() is returning a thread which is not threading.main_thread(), which is not expected at this point.

I added this because sometimes `tests_python.test_utilities.test_is_main_thread` fails due to that, so, this is to diagnose when that happens.